packages/fetchai/connections/p2p_libp2p/dht/dhtpeer/dhtpeer.go

```diff
@@ -537,35 +537,54 @@ func (dhtPeer *DHTPeer) lookupAddressDHT(address string) (peer.ID, error) {
 		Msgf("Querying for providers for cid %s...", addressCID.String())
 	ctx, cancel := context.WithTimeout(context.Background(), addressLookupTimeout)
 	defer cancel()
-	providers := dhtPeer.dht.FindProvidersAsync(ctx, addressCID, 1)
+	var elapsed time.Duration
+	var provider peer.AddrInfo
+	var connected bool = false
+	var s network.Stream
+
 	start := time.Now()
-	provider := <-providers
-	elapsed := time.Since(start)
-	for provider.ID == "" {
-		err = errors.New("didn't found any provider for address within timeout")
-		lerror(err).Str("op", "lookup").Str("addr", address).Msg("")
-		select {
-		default:
-			time.Sleep(200 * time.Millisecond)
-			providers = dhtPeer.dht.FindProvidersAsync(ctx, addressCID, 1)
-			provider = <-providers
+
+	for !connected {
+		providers := dhtPeer.dht.FindProvidersAsync(ctx, addressCID, 0)
```

Comment:
previously the third arg was 1, intentional?

Reply:
yes

Reply:
Would help to add a comment what the 0 is or Constantize it..another developer will be left guessing